FlowCal/mef.py

```diff
@@ -727,10 +729,16 @@ def get_transform_fxn(data_beads,
     # mef_channels and mef_values should be iterables.
     if hasattr(mef_channels, '__iter__'):
         mef_channels = list(mef_channels)
-        mef_values = np.array(mef_values).copy()
     else:
         mef_channels = [mef_channels]
-        mef_values = np.array([mef_values]).copy()
+        mef_values   = [mef_values]
+
+    # Ensure matching number of `mef_values` for all channels (this implies
+    # that the calibration beads have the same number of subpopulations for
+    # all channels).
+    if not np.all([len(mef_values_channel)==len(mef_values[0])
+                   for mef_values_channel in mef_values]):
+        raise ValueError("mef_values must be same length for all channels")
```

Comment:
I find the message a little confusing. It is not clear that it is the lengths of the innermost sequences in `mef_values` that should be the same.